File: framework/scala/src/main/scala/soda/scala/web/http/Part.scala
package soda.scala.web.http

import java.nio.charset.StandardCharsets

class Part {

  var contentDisposition: ContentDisposition = null

  var contentType: String = null

  var payload: Array[Byte] = Array.empty

  def getName: String = contentDisposition.name

  def bodyString = new String(payload, StandardCharsets.UTF_8)

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/web/http/Part.kt
package soda.kotlin.web.http

import java.nio.charset.StandardCharsets

class Part {

    var contentDisposition: ContentDisposition? = null

    var contentType: String = ""

    var payload = ByteArray(0)

    fun getName(): String = contentDisposition?.name ?: ""

    fun bodyString(): String = String(payload, StandardCharsets.UTF_8)

}